hack/logcollector/cmd/template.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/edgelesssys/constellation/v2/hack/logcollector/internal"
+	"github.com/spf13/cobra"
+)
+
+type infoMap map[string]string
+
+var defaultInfoMap = infoMap{
+	"is_debug_cluster": "false",
+}
+
+func newTemplateCmd() *cobra.Command {
+	templateCmd := &cobra.Command{
+		Use:   "template",
+		Short: "Templates filebeat and logstash configurations and prepares them for deployment",
+		Long:  `Templates filebeat and logstash configurations and prepares them for deployment by placing them in the specified directory.`,
+		RunE:  runTemplate,
+	}
+
+	templateCmd.Flags().String("dir", "", "Directory to place the templated configurations in (required)")
+	templateCmd.MarkFlagRequired("dir")
+	templateCmd.MarkFlagDirname("dir")
+	templateCmd.Flags().String("username", "", "OpenSearch username (required)")
+	templateCmd.MarkFlagRequired("username")
+	templateCmd.Flags().String("password", "", "OpenSearch password (required)")
+	templateCmd.MarkFlagRequired("password")
+	templateCmd.Flags().Int("port", 5045, "Logstash port")
+	templateCmd.Flags().StringToString("info", nil, "Additional fields for the Logstash pipeline in the format --info key1=value1,key2=value2,...")
```

Comment:
can you explain a little more what Logstash pipeline refers to? I guess these are metadata attributes (like tags) attached to the log entries?

Reply:
Exactly, the fields supplied here will be added as "tags" to the log entries. See https://www.elastic.co/guide/en/logstash/current/configuration.html and [us using them in the CI to set GitHub metadata](https://github.com/edgelesssys/constellation/pull/2080/files#diff-30c1e389a577e5a166b88e4dfd7b0c611f3305bf9bb9994ac2a79c5079d54978R25-R37)